### PR TITLE
Themes showcase: remove Beta badge for FSE themes

### LINKED ISSUE #58930: FSE Beta: Recommended and Full Site Editing tabs are duplicated in the Themes showcase
### What

**Current**

Screen Shot 2021-12-06 at 14 43 28

The "Full Site Editing (beta)" and "Recommended" tabs in the Theme showcase currently show the same things for FSE Beta sites. This seems unnecessary.

How about removing the extra tab and renaming "Recommended" so there's just one tab? This would be only for Sites that are FSE eligible (in the FSE Beta).

**Simplified**

image

### Why

To simplify the Themes showcase, and make it clear the Recommended themes are all FSE compatible themes.

### PR DESCRIPTION
**Pull request by creativecoder**
#### Changes proposed in this Pull Request

* Remove the "Beta" badge for block themes on FSE eligible sites
* Remove the "Full Site Editing" theme tab, since the themes are the same as what we show in the Recommended tab

Now that we are rapidly rolling out FSE to new users through the /start flow without any messaging about it being in beta, it seems better not to have "Beta" plastered all over the themes showcase.

#### Testing instructions

* On an FSE eligible site (you can create one through the wordpress.com/new flow, or horizon.wordpress.com/start)
* See that the themes showcase no longer shows the Beta badge on FSE themes
* And that the "Full Site Editing" theme tab is gone.

| **Before** | **After** |
| - | - |
| image | image |

Resolves https://github.com/Automattic/wp-calypso/issues/58930

### CODE REVIEW

File: client/components/theme/index.jsx [hunk 1]
```diff
@@ -145,7 +143,7 @@ export class Theme extends Component {
 	};
 
 	render() {
-		const { active, blockEditorSettings, price, theme, translate, upsellUrl } = this.props;
```
Comment (zaguiini) [inline client/components/theme/index.jsx:148]: Now that we're not using `blockEditorSettings` anymore, we can remove the HOC below and the prop type declaration.

File: client/my-sites/themes/theme-showcase.jsx [hunk 2]
```diff
@@ -284,9 +270,6 @@ class ThemeShowcase extends Component {
 				return true;
 			case this.tabFilters.MYTHEMES.key:
 				return this.props.isJetpackSite;
-			case this.tabFilters.FSE.key:
-				// Display FSE tab if the Site Editor is active for the site.
-				return this.props.blockEditorSettings?.is_fse_eligible;
```
Comment (zaguiini) [inline client/my-sites/themes/theme-showcase.jsx:289]: The same thing here, we can get rid of `blockEditorSettings` for this file.

**Review by zaguiini — COMMENTED** [review]
There's also:

- `client/my-sites/theme/main.jsx`;
- `client/my-sites/themes/current-theme/index.tsx`...

displaying the `Beta` badge. Will we get rid of it there as well?

We probably can un-HOC `client/my-sites/themes/theme-showcase.jsx` too because the prop is not being used there anymore, nor in the children that gets props from this component.

File: client/components/theme/index.jsx (hunk 1 above)
Comment (jeyip) [reply client/components/theme/index.jsx:148]: Done 👍

**Comment by jeyip** [comment]
@creativecoder I made the changes that @zaguiini suggested. Could you do another smoke test? Things look good on my end.

**Comment by jeyip** [comment]
## Testing
### Requirements
- [x]  The "Full Site Editing" theme tab is gone.
- [x] The themes cards in the themes showcase no longer show the Beta badge on FSE themes
- [x] The current theme card in the themes showcase no longer shows the Beta badge for an FSE theme
Screen Shot 2022-02-01 at 10 51 39 AM

- [x] The theme info page no longer shows a beta badge for an FSE theme
Screen Shot 2022-02-01 at 10 52 04 AM

### Browsers
- [x] Edge
- [x] Firefox
- [x] Chrome

**Comment by creativecoder** [comment]
Sorry about the mass ping, y'all--I borked a rebase.

**Comment by creativecoder** [comment]
Thanks for the suggestions @zaguiini and for the changes @jeyip . I've also removed the use of another blockEditorSettings HOC, deleted a couple of now unused prop types, and fixed a test. This is ready for a final review.

**Review by jeyip — APPROVED** [review]
LGTM 🚢

**Pull request merged**